src/poprox_recommender/recommenders.py

```diff
@@ -58,7 +82,45 @@ def select_articles(
     else:
         wanted = (topk, recs)
 
-    return pipeline.run_all(*wanted, candidate=candidate_articles, clicked=clicked_articles, profile=interest_profile)
+    pipeline_state = pipeline.run_all(
```

Comment:
We want `pipeline.run_all()` to execute your components too, which means your components need to be added to the pipeline. Here's an example of how pipelines are built: https://github.com/CCRI-POPROX/poprox-recommender/blob/main/src/poprox_recommender/recommenders/configurations/mmr.py

When you call `builder.add_component()`, you provide information about what class to use and how to wire up the inputs to other components in the pipeline. The names of the arguments should match the names on your `__call__()` methods.